core/os/device/device.proto

```diff
@@ -56,6 +56,21 @@ enum OSKind {
   Stadia = 5;
 }
 
+// Incompatibility list possible reasons why a device is not compatible to
+// replay a given trace.
+enum Incompatibility {
+  Compatible = 0;  // Default, for compatible devices
+  ABIOS = 1;
+  ABIArchitecture = 2;
+  ABIMemoryLayout = 3;
```

Comment:
Remove the `ABI` part of these enums.

Reply:
We can't, it leads to errors like:

```
core/os/device/device.proto:63:3: "OS" is already defined in "device".
core/os/device/device.proto:63:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "OS" must be unique within "device", not just within "Incompatibility".
core/os/device/device.proto:64:3: "Architecture" is already defined in "device".
core/os/device/device.proto:64:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "Architecture" must be unique within "device", not just within "Incompatibility".
core/os/device/device.proto:65:3: "MemoryLayout" is already defined in "device".
core/os/device/device.proto:65:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "MemoryLayout" must be unique within "device", not just within "Incompatibility".
```

That's why I added the "ABI" prefix.

But I think that by renaming this enum to `ReplayCompatibility` as you suggest, we may use enum names like `BadOS` or `InvalidOS`.

Reply:
🙄  I see. I'd prefer something like `Incompatable<foo>` or `<foo>Mismatch`. The OS is neither bad nor invalid. :)

Reply:
:+1:

Reply:
Renamed.